### PR TITLE
[FEATURE] add enums and protobufs for new cs2 ranks

### PR DESCRIPTION
**Pull request by Tom-jm03**
## Summary

Add cs2 protobuffs and enums

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: steam/ext/csgo/enums.py [hunk 1]
```diff
@@ -98,7 +98,25 @@ class ItemOrigin(IntEnum):
     LevelUpReward             = 24
     Max                       = 25
 
+class PerMapRankEnum(IntEnum):
```
Comment (Gobot1234) [inline steam/ext/csgo/enums.py:101]: Drop the enum suffix from these 2 classes

File: steam/ext/csgo/enums.py [hunk 2]
```diff
@@ -98,7 +98,25 @@ class ItemOrigin(IntEnum):
...
+    MapID                     = 1
+    RANK_ID                   = 2
+    WINS                      = 3
```
Comment (Gobot1234) [inline steam/ext/csgo/enums.py:104]:
```suggestion
    MapID                     = 1
    RANK_ID                   = 2
    WINS                      = 3
```
Consistent naming here is required

File: steam/ext/csgo/protobufs/ranks.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+import betterproto
+
+from ....protobufs.msg import GCProtobufMessage
+from ..enums import EMsg
+
+
+@dataclass(eq=False, repr=False)
+class PlayerRankingInfo(betterproto.Message):
```
Comment (Gobot1234) [inline steam/ext/csgo/protobufs/ranks.py:10]:
```suggestion
class PlayerRankingInfo(GCProtobufMessage, msg=...):  # something should go here to link it
```

File: steam/ext/csgo/protobufs/ranks.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    wins: int = betterproto.uint32_field(3)
+    rank_change: float = betterproto.float_field(4)
+    rank_type_id: int = betterproto.uint32_field(6)
+    tv_control: int = betterproto.uint32_field(7)
+    rank_window_stats: int = betterproto.uint64_field(8)
+    leaderboard_name: str = betterproto.string_field(9)
+    rank_if_win: int = betterproto.uint32_field(10)
+    rank_if_lose: int = betterproto.uint32_field(11)
+    rank_if_tie: int = betterproto.uint32_field(12)
+    per_map_rank: "list[PerMapRank]" = betterproto.message_field(13, repeated=True)
```
Comment (Gobot1234) [inline steam/ext/csgo/protobufs/ranks.py:22]:
```suggestion
    per_map_rank: "list[PerMapRank]" = betterproto.message_field(13)
```
This isn't gonna work with betterproto, repeated isn't a kwarg it takes. If you're gonna use co-pilot to generate these you'll need to take a closer look at the other examples. Ideally if you've never used them before you should be using the protoc CLI (see https://github.com/danielgtaylor/python-betterproto?tab=readme-ov-file#compiling-proto-files)

**Review by Gobot1234 — CHANGES_REQUESTED** [review]
Thanks for this, 

Currently, it is missing a method to actually make use of these changes so this wouldn't be useable for anyone

File: steam/ext/csgo/enums.py (hunk 1 above)
Comment (Tom-jm03) [reply steam/ext/csgo/enums.py:101]: Done.

File: steam/ext/csgo/enums.py (hunk 2 above)
Comment (Tom-jm03) [reply steam/ext/csgo/enums.py:104]: Done.

File: steam/ext/csgo/protobufs/ranks.py (hunk 3 above)
Comment (Tom-jm03) [reply steam/ext/csgo/protobufs/ranks.py:10]: Done.

File: steam/ext/csgo/protobufs/ranks.py (hunk 4 above)
Comment (Tom-jm03) [reply steam/ext/csgo/protobufs/ranks.py:22]: Done.

**Comment by Gobot1234** [comment]
Can you push your changes if you've made them please?

**Comment by Tom-jm03** [comment]
> Can you push your changes if you've made them please?

Done. But it is still missing the method, im working on that.

File: steam/ext/csgo/enums.py [hunk 5]
```diff
@@ -98,7 +98,25 @@ class ItemOrigin(IntEnum):
     LevelUpReward             = 24
     Max                       = 25
 
+class PerMapRankEnum:
```
Comment (Gobot1234) [inline steam/ext/csgo/enums.py:101]: These should still be IntEnum subclasses, I meant change the names of the classes to remove Enum from the end

File: steam/ext/csgo/protobufs/ranks.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+import betterproto
+
+from ....protobufs.msg import GCProtobufMessage
+from ..enums import EMsg
+
+
+@dataclass(eq=False, repr=False)
```
Comment (Gobot1234) [inline steam/ext/csgo/protobufs/ranks.py:9]: This decorator isn't necessary

**Review by Gobot1234 — CHANGES_REQUESTED** [review]

File: steam/ext/csgo/enums.py (hunk 5 above)
Comment (Tom-jm03) [reply steam/ext/csgo/enums.py:101]: Oh okay, my bad.

File: steam/ext/csgo/protobufs/ranks.py (hunk 6 above)
Comment (Tom-jm03) [reply steam/ext/csgo/protobufs/ranks.py:9]: What about the second decorator?  Also not necessary? ` @dataclass(eq=False, repr=False)`

Comment (Gobot1234) [reply steam/ext/csgo/protobufs/ranks.py]: No the second one is because it subclasses betterproto.Message directly

Comment (Tom-jm03) [reply steam/ext/csgo/protobufs/ranks.py]: Do we actually need a new method?
We already got this one:

```py
    async def fetch_user_csgo_profile(self, user_id: int) -> cstrike.PlayersProfile:
        await self.ws.send_gc_message(cstrike.ClientRequestPlayersProfile(account_id=user_id, request_level=32))
        return await self.ws.gc_wait_for(
            cstrike.PlayersProfile,
            check=lambda msg: msg.account_profiles[0].account_id == user_id,
        )
```

**Comment by Tom-jm03** [comment]
The method is now finished.

File: steam/ext/csgo/enums.py [hunk 7]
```diff
@@ -98,7 +98,25 @@ class ItemOrigin(IntEnum):
     LevelUpReward             = 24
     Max                       = 25
 
+class PerMapRank(IntEnum):
+    MapID                     = 1
+    RankID                    = 2
+    Wins                      = 3
```
Comment (Gobot1234) [inline steam/ext/csgo/enums.py:105]:
```suggestion


```

File: steam/ext/csgo/enums.py [hunk 8]
```diff
@@ -98,7 +98,25 @@ class ItemOrigin(IntEnum):
...
+    RankChange                  = 4
+    RankeTypeID                 = 5
+    TVControl                   = 6
+    RankWindowStats             = 7
+    LeaderboardName             = 8
+    RankIfWin                   = 9
+    RankIfLose                  = 10
+    RankIfTie                   = 11
+    PerMapRank                  = 12
+    LeaderboardNameStatus       = 13
```
Comment (Gobot1234) [inline steam/ext/csgo/enums.py:119]:
```suggestion
    LeaderboardNameStatus       = 13

```

File: steam/ext/csgo/state.py [hunk 9]
```diff
@@ -202,13 +202,20 @@ async def update_backpack(self, *gc_items: base.Item, is_cache_subscribe: bool =
...
             check=lambda msg: msg.account_profiles[0].account_id == user_id,
         )
 
+    #    async def fetch_user_cs2_profile(self, user_id: int) -> cstrike.PlayersProfile:
+    #        await self.ws.send_gc_message(cstrike.ClientRequestPlayersProfilecs2(account_id=user_id, request_level=32))
+    #        return await self.ws.gc_wait_for(
+    #            cstrike.PlayersProfile,
+    #            check=lambda msg: msg.account_profiles[0].account_id == user_id,
+    #        )
+
```
Comment (Gobot1234) [inline steam/ext/csgo/state.py:218]:
```suggestion
```

File: steam/ext/csgo/protobufs/cstrike.py [hunk 10]
```diff
@@ -718,6 +731,13 @@ class ClientRequestPlayersProfile(GCProtobufMessage, msg=EMsg.ClientRequestPlaye
     request_level: int = betterproto.uint32_field(4)
 
 
+# class ClientRequestPlayersProfilecs2(GCProtobufMessage, msg=EMsg.ClientRequestPlayersProfilecs2):
+#    request_id_deprecated: int = betterproto.uint32_field(1)
+#    account_ids_deprecated: list[int] = betterproto.uint32_field(2)
+#    account_id: int = betterproto.uint32_field(3)
+#    request_level: int = betterproto.uint32_field(4)
+
+
```
Comment (Gobot1234) [inline steam/ext/csgo/protobufs/cstrike.py:740]:
```suggestion
```

**Review by Gobot1234 — APPROVED** [review]

File: steam/ext/csgo/models.py [hunk 11]
```diff
@@ -245,9 +245,17 @@ def __init__(self, user: UserT, proto: cstrike.MatchmakingClientHello | cstrike.
...
+        self.rank_if_lose: int | None = proto.rank_if_lose
+        self.rank_if_tie: int | None = proto.rank_if_tie
+        self.rank_type_id: int | None = getattr(proto, "rank_type_id", None)
+        self.rank_change: int | None = getattr(proto, "rank_change", None)
+        self.rank_window_stats: int | None = getattr(proto, "rank_window_stats", None)
+        self.leaderboard_name_status: int | None = getattr(proto, "leaderboard_name_status", None)
         self.level = Level(proto.player_level)
         self.xp_bonus_flags = proto.player_xp_bonus_flags
         self.rankings = Rank(proto.rankings)
+        self.per_map_rank: list | None = proto.per_map_rank
```
Comment (Gobot1234) [inline steam/ext/csgo/models.py:258]: This annotation is not necessary

**Comment by Tom-jm03** [comment]
I think we can close this pr, as you can already get the cs2 things without any updates...

**Pull request closed without merging**